From 5cbb7051f64cb6590292196340d0360fa1abf8d5 Mon Sep 17 00:00:00 2001
From: Pierre-Hugues Husson <phh@phh.me>
Date: Tue, 10 Mar 2020 23:26:55 +0100
Subject: [PATCH] Use BKK's hwcomposer hacks to properly display FOD layers

Insecure hack: Set FOD layers to magical Z values to trigger
OPPO/Realme/Oneplus' whole blitter behaviours:
- reports touch event to fingerprint driver
- removes dim on touched layer

Change-Id: I1d924b1e5c2389c8df17174c079cee8ef2380112

diff --git a/services/surfaceflinger/CompositionEngine/src/OutputLayer.cpp b/services/surfaceflinger/CompositionEngine/src/OutputLayer.cpp
index ebf651ee8..c7c900d82 100644
--- a/services/surfaceflinger/CompositionEngine/src/OutputLayer.cpp
+++ b/services/surfaceflinger/CompositionEngine/src/OutputLayer.cpp
@@ -378,6 +378,15 @@ void OutputLayer::writeOutputDependentGeometryStateToHWC(
         z = getFodZOrder(z, true);
     }
 
+    if(strstr(getLayerFE().getDebugName(), "Fingerprint on display") != nullptr) {
+        ALOGE("Found fingerprint on display!");
+        z = 0x41000031;
+    }
+
+    if(strstr(getLayerFE().getDebugName(), "Fingerprint on display.touched") != nullptr) {
+        ALOGE("Found fingerprint on display touched!");
+        z = 0x41000033;
+    }
     if (auto error = hwcLayer->setZOrder(z); error != hal::Error::NONE) {
         ALOGE("[%s] Failed to set Z %u: %s (%d)", getLayerFE().getDebugName(),
               outputDependentState.z, to_string(error).c_str(), static_cast<int32_t>(error));
-- 
2.25.1

